tests/test_github_bookmarks.py

```diff
@@ -31,7 +33,9 @@ def calculated_states_by_stream(self, current_state, synced_records, replication
             state_as_datetime = dateutil.parser.parse(state_value)
 
             days, hours, minutes = timedelta_by_stream[stream]
-            calculated_state_as_datetime = state_as_datetime - datetime.timedelta(days=days, hours=hours, minutes=minutes)
+
+            start_date_as_datetime = dateutil.parser.parse(start_date)
+            calculated_state_as_datetime = start_date_as_datetime + datetime.timedelta(days=days, hours=hours, minutes=minutes)
```

Comment:
Doesn't this change the expectations in strange ways? Going off of the start date instead of the first sync's final bookmark?

It's a bit tough to follow, but it seems like this was attempting to always re-sync some data on the second sync by starting from the first sync's successful bookmarking and then looking back X days. This now looks forward from start_date, which possibly gets data, but does it still live up to the existing expectations?

Reply:
I'll check with QA about this.

My understanding of the bookmark tests has been that it verifies that the tap can sync from any bookmark. And the test use a bookmark that excludes at least one record to show that it works